src/s3/commands/presignedURL.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SignedUrlRequest } from '../../shared/clients/s3Client'
+import { Env } from '../../shared/vscode/env'
+import { S3FileNode } from '../explorer/s3FileNode'
+import { Window } from '../../shared/vscode/window'
+import { addCodiconToString } from '../../shared/utilities/textUtilities'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { COPY_TO_CLIPBOARD_INFO_TIMEOUT_MS } from '../../shared/constants'
+import { validateTime } from '../util'
+
+export async function presignedURLCommand(
+    node: S3FileNode,
+    window = Window.vscode(),
+    env = Env.vscode()
+): Promise<void> {
+    const time = await promptTime(node.file.key, window)
+    const s3Client = node.s3
+
+    const request: SignedUrlRequest = {
+        bucketName: node.bucket.name,
+        key: node.file.key,
+        operation: 'getObject',
+        time: time,
+    }
+
+    const url = await s3Client.getSignedUrl(request)
+
+    await copyUrl(url, window, env)
+}
+
+export async function promptTime(fileName: string, window = Window.vscode()): Promise<number> {
+    const timeStr = await window.showInputBox({
+        value: '15',
+        prompt: localize('AWS.s3.promptTime.prompt', 'Please enter the valid time (in minutes) for ${0} URL', fileName),
+        placeHolder: localize('AWS.s3.promptTime.placeHolder', 'Defaults to 15 minutes'),
+        validateInput: validateTime,
+    })
+
+    const time = Number(timeStr)
+
+    if (isNaN(time) || time < 0) {
+        return 15
+    }
```

Comment:
The validation function shouldn't let a user return a bad value. If they do, I'd rather throw an error than return 15 to reduce ambiguity.